mobile/lib/extensions/collection_extensions.dart

```diff
@@ -39,3 +41,52 @@ extension IntListExtension on Iterable<int> {
     return list;
   }
 }
+
+extension AssetListExtension on Iterable<Asset> {
+  Iterable<Asset> remoteOnly({
+    void Function()? errorCallback,
+  }) {
+    final bool onlyRemote = every((e) => e.isRemote);
+    if (!onlyRemote) {
+      if (errorCallback != null) errorCallback();
+      return where((a) => a.isRemote);
+    }
+    return this;
+  }
+
+  Iterable<Asset> ownedOnly(
+    User? owner, {
+    void Function()? errorCallback,
+  }) {
+    if (owner == null) return [];
+    final userId = owner.isarId;
+    final bool onlyOwned = every((e) => e.ownerId == userId);
+    if (!onlyOwned) {
+      if (errorCallback != null) errorCallback();
+      return where((a) => a.ownerId == userId);
+    }
+    return this;
+  }
+
+  Iterable<Asset> writableOnly({
+    void Function()? errorCallback,
+  }) {
+    final bool onlyWritable = every((e) => !e.isReadOnly);
+    if (!onlyWritable) {
+      if (errorCallback != null) errorCallback();
+      return where((a) => !a.isReadOnly);
+    }
+    return this;
+  }
+
+  Iterable<Asset> liveOnly({
```

Comment:
I assume live here is LivePhotos?

Reply:
It was actually to filter out non offline files, so updated the method name to `nonOfflineOnly` to better indicate the same